### PR TITLE
ISPN-6577 Refactor server indexing configuration to its own subresource

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-6577

### CODE REVIEW

**Comment by tristantarrant** [comment]
@ryanemerson this has impact on the indexing configuration for the console. Indexing properties are no longer direct children of the cache but of the indexing=INDEXING resource (similar to e.g. eviction=EVICTION)

**Comment by ryanemerson** [comment]
@tristantarrant https://github.com/infinispan/infinispan-management-console/pull/95 is up. So we will integrate once @vblagoje has approved the fix and this commit has been integrated.

**Comment by anistor** [comment]
Looking ..

**Comment by gustavonalle** [comment]
Looks good to me, but we could use a test to check that index is really enabled when using the config from the JIRA

**Comment by gustavonalle** [comment]
@tristantarrant What about backporting this to 8.2.x?

File: server/integration/infinispan/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties [hunk 1]
```diff
@@ -324,7 +320,16 @@ datagrid-infinispan.expiration.lifespan=Maximum lifespan of a cache entry, after
 datagrid-infinispan.expiration.interval=Interval (in milliseconds) between subsequent runs to purge expired entries from memory and any cache stores. If you wish to disable the periodic eviction process altogether, set wakeupInterval to -1.
 datagrid-infinispan.expiration.add=Adds an expiration configuration element to the cache.
 datagrid-infinispan.expiration.remove=Removes an expiration configuration element from the cache.
+#
+datagrid-infinispan.indexing=The cache eviction configuration.
```
Comment (anistor) [inline server/integration/infinispan/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: You definitely meant to write 'cache indexing', not 'cache eviction'.

File: server/integration/infinispan/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties [hunk 2]
```diff
@@ -146,16 +146,12 @@ datagrid-infinispan.cache.start=The cache start mode, which can be EAGER (immedi
...
-datagrid-infinispan.cache.indexing=If enabled, entries will be indexed when they are added to the cache. Indexes will be updated as entries change or are removed.
 datagrid-infinispan.cache.jndi-name=The jndi-name to which to bind this cache instance.
 datagrid-infinispan.cache.module=The module whose class loader should be used when building this cache's configuration.
-datagrid-infinispan.cache.auto-config=If enabled, will configure indexing automatically based on cache type
-datagrid-infinispan.cache.indexing-properties=Properties to control indexing behaviour
 datagrid-infinispan.cache.remove=Remove a cache from this container.
 datagrid-infinispan.cache.remote-cache=The name of the remote cache that backups data here.
 datagrid-infinispan.cache.remote-site=The name of the remote site containing the cache that backups data here.
-datagrid-infinispan.cache.template=Whether the configuration is a template or for a concrete cache.
+datagrid-infinispan.cache.template=Whether the configuration is a template or for a concrete cache.dex
```
Comment (anistor) [inline server/integration/infinispan/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: ```cache.dex``` ???

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/IndexingConfigurationResource.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2012, Red Hat, Inc., and individual contributors
```
Comment (anistor) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/IndexingConfigurationResource.java]: 2016

**Comment by tristantarrant** [comment]
@gustavonalle @anistor Now comes complete with a test for checking that the cache has the right indexing mode.

File: server/integration/infinispan/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties [hunk 4]
```diff
@@ -324,7 +320,16 @@ datagrid-infinispan.expiration.lifespan=Maximum lifespan of a cache entry, after
 datagrid-infinispan.expiration.interval=Interval (in milliseconds) between subsequent runs to purge expired entries from memory and any cache stores. If you wish to disable the periodic eviction process altogether, set wakeupInterval to -1.
 datagrid-infinispan.expiration.add=Adds an expiration configuration element to the cache.
 datagrid-infinispan.expiration.remove=Removes an expiration configuration element from the cache.
+#
+datagrid-infinispan.indexing=The cache indexing configuration.
+datagrid-infinispan.indexing.auto-config=If enabled, will configure indexing automatically based on cache type
+datagrid-infinispan.indexing.indexing=If enabled, entries will be indexed when they are added to the cache. Indexes will be updated as entries change or are removed.
+datagrid-infinispan.indexing.indexed-entities=The fully qualified names of the indexed entity classes.
+datagrid-infinispan.indexing.indexing-properties=Properties to control indexing behaviour
+datagrid-infinispan.indexing.add=Adds an eviction configuration element to the cache.
```
Comment (anistor) [inline server/integration/infinispan/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: Replace `eviction` with `indexing`

**Comment by tristantarrant** [comment]
Done

File: server/integration/infinispan/src/test/resources/org/jboss/as/clustering/infinispan/subsystem/indexing-inheritance.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<subsystem xmlns="urn:infinispan:server:core:9.0">
+    <cache-container name="indexing" default-cache="local">
```
Comment (anistor) [inline server/integration/infinispan/src/test/resources/org/jboss/as/clustering/infinispan/subsystem/indexing-inheritance.xml]: A cache named `local` is never defined. Is it ok?

Comment (tristantarrant) [inline server/integration/infinispan/src/test/resources/org/jboss/as/clustering/infinispan/subsystem/indexing-inheritance.xml]: quite irrelevant :)

**Comment by anistor** [comment]
Looks good to me.

**Comment by anistor** [comment]
Pulling...

**Comment by anistor** [comment]
Integrated in master. Thanks @tristantarrant !

**Pull request merged**